### PR TITLE
[Custom Metrics] Improve ray runtime context.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

- Add new methods.
- Refine some code.
- Improve testing (make it faster).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/runtime_context.py [hunk 1]
```diff
@@ -11,28 +11,107 @@ def __init__(self, worker):
         assert worker is not None
         self.worker = worker
 
+    def get(self):
+        """Get a dictionary of the current_context.
+
+        The value contains None if the field is not available.
+        For example, if you invoke this method from a driver,
+        task_id or actor_id is not available. In this case,
+        the dictionary will contain {"actor_id": None, "task_id": None}
```
Comment (edoakes) [inline python/ray/runtime_context.py:20]: Shouldn't this just not contain those keys? That would probably work better for using w/ the metrics API, right?

File: python/ray/tests/test_runtime_context.py [hunk 2]
```diff
@@ -65,10 +59,17 @@ def current_job_id(self):
         def current_actor_id(self):
             return ray.get_runtime_context().current_actor_id
 
+    @ray.remote
+    def f():
+        assert ray.get_runtime_context().current_actor_id is None
+        assert ray.get_runtime_context().current_task_id is not None
+        assert ray.get_runtime_context().current_node_id is not None
+        assert ray.get_runtime_context().current_job_id is not None
```
Comment (edoakes) [inline python/ray/tests/test_runtime_context.py:67]: should we just remove `current` from all of these? it seems sort of redundant

Comment (rkooo567) [reply python/ray/tests/test_runtime_context.py]: Got it. Will fix.

File: python/ray/runtime_context.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/runtime_context.py]: Hmm that's a good point. You are saying we can do sth like

```
metric.record(1, tags=runtime_context().get())
```

right? I think it makes sense. Let me fix it.

File: python/ray/runtime_context.py [hunk 3]
```diff
@@ -11,28 +11,102 @@ def __init__(self, worker):
...
+        Returns:
+            dict: Dictionary of the current context.
+        """
+        context = {
+            "job_id": self.job_id,
+            "node_id": self.node_id,
+            "task_id": self.task_id,
+            "actor_id": self.actor_id
+        }
+        return {
```
Comment (rkooo567) [inline python/ray/runtime_context.py:26]: cc @edoakes Lmk if you like this style better.

```
result = {}
job_id = self.job_id
if job_id:
    result["job_id"] = job_id
```

**Comment by rkooo567** [comment]
cc @edoakes  This addressed all reviews!

File: python/ray/runtime_context.py (hunk 3 above)
Comment (edoakes) [reply python/ray/runtime_context.py:26]: This looks good

Comment (edoakes) [inline python/ray/runtime_context.py:26]: Maybe just leave a comment saying "Remove keys whose values are None." or similar

File: python/ray/tests/test_runtime_context.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 import sys
 
 
-def test_was_current_actor_reconstructed():
+def test_was_current_actor_reconstructed(shutdown_only):
```
Comment (edoakes) [inline python/ray/tests/test_runtime_context.py:8]: nice :D

**Review by edoakes — APPROVED** [review]

**Pull request merged**